Compilation error in vortex-array@0.56.0

### Discussed in https://github.com/vortex-data/vortex/discussions/5671

<div type='discussions-op-text'>

<sup>Originally posted by **duarten** December 10, 2025</sup>
### Issue Description

With Rust 1.89.0 and above:

```
error[E0282]: type annotations needed
  --> /Users/duarte.nunes/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/vortex-array-0.56.0/src/expr/display.rs:30:21
   |
30 |                     Ok(Tree::new(format!("{}: {}", name, child_tree.root))
   |                     ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
   |
help: consider specifying the generic arguments
   |
30 |                     Ok::<Tree<std::string::String>, E>(Tree::new(format!("{}: {}", name, child_tree.root))
   |                       ++++++++++++++++++++++++++++++++

error[E0283]: type annotations needed
  --> /Users/duarte.nunes/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/vortex-array-0.56.0/src/expr/display.rs:30:21
   |
29 |                     let child_tree = make_tree(child)?;
   |                                                      - type must be known at this point
30 |                     Ok(Tree::new(format!("{}: {}", name, child_tree.root))
   |                     ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
   |
   = note: multiple `impl`s satisfying `_: From<std::fmt::Error>` found in the following crates: `idna`, `log`, `serde_fmt`, `value_bag`:
           - impl From<std::fmt::Error> for idna::punycode::PunycodeEncodeError;
           - impl From<std::fmt::Error> for idna::uts46::ProcessingError;
           - impl From<std::fmt::Error> for log::kv::Error;
           - impl From<std::fmt::Error> for serde_fmt::Error;
           - impl From<std::fmt::Error> for value_bag::error::Error;
help: consider specifying the generic arguments
   |
30 |                     Ok::<Tree<std::string::String>, E>(Tree::new(format!("{}: {}", name, child_tree.root))
   |                       ++++++++++++++++++++++++++++++++

error[E0283]: type annotations needed
  --> /Users/duarte.nunes/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/vortex-array-0.56.0/src/expr/display.rs:30:21
   |
30 |                     Ok(Tree::new(format!("{}: {}", name, child_tree.root))
   |                     ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
...
35 |             Ok(Tree::new(node_name).with_leaves(child_trees?))
   |                                                            - type must be known at this point
   |
   = note: multiple `impl`s satisfying `std::fmt::Error: From<_>` found in the following crates: `core`, `serde_fmt`:
           - impl From<serde_fmt::Error> for std::fmt::Error;
           - impl<T> From<T> for T;
help: consider specifying the generic arguments
   |
30 |                     Ok::<Tree<std::string::String>, E>(Tree::new(format!("{}: {}", name, child_tree.root))
```

### Expected Behavior

Compilation to succeed.

### Actual Behavior

Compilation fails.

### Reproduction Steps

Try to compile `vortex-array`@0.56.0

### OS Version Information

macOS 15

### I acknowledge that:

- [X] I have searched the Vortex repository (both open and closed Discussions and Issues) and confirm this is not a duplicate of an existing issue or discussion.
- [X] I have checked the "Preview" tab on all text fields to ensure that everything looks right, and have wrapped all configuration and code in code blocks with a group of three backticks (` ``` `) on separate lines.</div>

--- a/vortex-array/src/expr/display.rs
+++ b/vortex-array/src/expr/display.rs
@@ -23,13 +23,15 @@ impl Display for DisplayTreeExpr<'_> {
             let child_names = (0..expr.children().len()).map(|i| expr.child_name(i));
             let children = expr.children();
 
-            let child_trees: Result<Vec<Tree<String>>, _> = children
+            let child_trees: Result<Vec<Tree<String>>, std::fmt::Error> = children
                 .iter()
                 .zip(child_names)
                 .map(|(child, name)| {
                     let child_tree = make_tree(child)?;
-                    Ok(Tree::new(format!("{}: {}", name, child_tree.root))
-                        .with_leaves(child_tree.leaves))
+                    Ok::<Tree<String>, std::fmt::Error>(
+                        Tree::new(format!("{}: {}", name, child_tree.root))
+                            .with_leaves(child_tree.leaves),
+                    )
                 })
                 .collect();
 

